Library/ViewModels/CommentRepliesViewModel.swift

```diff
@@ -94,18 +97,35 @@ public final class CommentRepliesViewModel: CommentRepliesViewModelType,
         return (url, canPostComment, false, inputAreaBecomeFirstResponder)
       }
 
-    let repliesEvent = rootComment
-      .switchMap { comment in
-        AppEnvironment.current.apiService.fetchCommentReplies(
-          query: commentRepliesQuery(withCommentId: comment.id)
-        )
-        .materialize()
-      }
+    let totalCountProperty = MutableProperty<Int>(0)
+
+    // TODO: Handle isLoading from here
+    let (replies, _, _, _) = paginate(
+      requestFirstPageWith: rootComment,
+      requestNextPageWhen: self.viewMoreRepliesCellWasTappedProperty.signal,
+      clearOnNewRequest: true,
+      valuesFromEnvelope: { [totalCountProperty] envelope -> [Comment] in
+        totalCountProperty.value = envelope.totalCount
+        return envelope.replies
+      },
+      cursorFromEnvelope: { envelope in
+        (envelope.comment, envelope.cursor)
+      },
+      requestFromParams: { comment in
+        AppEnvironment.current.apiService
+          .fetchCommentReplies(query: commentRepliesQuery(withCommentId: comment.id))
+      },
+      requestFromCursor: {
+        AppEnvironment.current.apiService
+          .fetchCommentReplies(query: commentRepliesQuery(withCommentId: $0.0.id, before: $0.1))
```

Comment:
Maybe can make this a little clearer to read like in `requestFromParams`. `commentAndCursor.comment.id` and `commentAndCursor.cursor`

Reply:
Good catch.